src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/TypePreinit.cs

```diff
@@ -612,6 +612,12 @@ private Status TryScanMethod(MethodIL methodIL, Value[] parameters, Stack<Method
 
                                 instance = new ReadOnlySpanValue(readOnlySpanElementType, bytes, byteOffset, byteLength);
                             }
+                            else if (readOnlySpanElementType != null && ctorSig.Length == 1 && ctorSig[0].IsArray
```

Comment:
It took me some extra time to see that `readOnlySpanElementType` is set in the previous if condition. Would it be better to have all span ctors under the same if check? Something like:

```
else if ((TryGetSpanElementType(owningType, isReadOnlySpan: true, out MetadataType readOnlySpanElementType)
    || TryGetSpanElementType(owningType, isReadOnlySpan: false, out readOnlySpanElementType))
{
    if (ctorSig.Length == 1 ...)
    {
...
    }
    else if (ctorSig.Length == 2 ...)
    {
...
    }
}
```

Reply:
Yeah, I don't like it either, but we'd need a `goto` to line 623 (in the new file, or 617 in the before version). If it's a newobj on readonly span that we don't intrinsically recognize, it needs to go to the "usual interpreter handling".